src/abstractdataframe/abstractdataframe.jl

```diff
@@ -63,10 +63,21 @@ abstract type AbstractDataFrame end
 
 """
     names(df::AbstractDataFrame)
+    names(df::AbstractDataFrame, cols)
 
     Return a `Vector{Symbol}` of names of columns contained in `df`.
+
+    If `cols` column selector argument is passed then restrict returned
+    column names to those matching the selector
+    (this is useful with regular expressions, `All`, `Not`, and `Between`).
```

Comment:
```suggestion
    (this is useful in particular with regular expressions, `Not`, and `Between`).
```
(`All` doesn't sound too useful, and passing indices could be useful in some cases)

Also, don't we generally repeat the list of all possible selectors in all docstrings?

Reply:
OK - I will fix this